## Выбираем лучшие настройки для тикеров

In [1]:
%reload_ext autoreload
%autoreload 2

In [23]:
from bot.test import TestAlgorithm, TestAccAlgorithm
from app.cache import LocalCache
from bot.vis import TaskProgress
from app.config import RunConfig, AccConfig

test_configs = [
    {
        'acc': AccConfig.from_repr_string('t2 [0] |u0.2 d0|'),
        'bots':  [
            RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
        ]
    },
    # {
    #     'acc': AccConfig.from_repr_string('t2 [0] |u0.25 d0|'),
    #     'bots':  [
    #         RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
    #     ]
    # },
    {
        'acc': AccConfig.from_repr_string('t3 [0] |u0.3 d0|'),
        'bots':  [
            RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
        ]
    },
    # {
    #     'acc': AccConfig.from_repr_string('t3 [0] |u0.35 d0|'),
    #     'bots':  [
    #         RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
    #     ]
    # },
    # {
    #     'acc': AccConfig.from_repr_string('t5 [0] |u0.4 d0|'),
    #     'bots':  [
    #         RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
    #         # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
    #     ]
    # },
    {
        'acc': AccConfig.from_repr_string('t5 [0] |u0.5 d0|'),
        'bots':  [
            RunConfig.from_repr_string('MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            RunConfig.from_repr_string('RNFT+ 2/fan7:0/2 x l1 x 0.6(+x0.2)¤'),
            # RunConfig.from_repr_string('MTLR+ 2/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0|'),
        ]
    }
]

def run_test(config):
    bot_alg_list = []
    for bot_cfg in config['bots']:
        bot_alg_list.append(TestAlgorithm(do_printing=False, config=bot_cfg, use_cache=True))
        
    test_acc_alg = TestAccAlgorithm(
        do_printing=False, 
        config=config['acc'],
        bot_alg_list=bot_alg_list,
    )
    
    return test_acc_alg.test(
        last_test_date='2024-10-24',
        test_days_num=365,
        # last_test_date='2024-12-31',
        # test_days_num=365,
    )

# unique_configs = set(test_configs)
unique_configs = test_configs

results = []
LocalCache.clear()
progress = TaskProgress(len(unique_configs))

for cfg in test_configs:
    res = run_test(cfg)
    if res:
        results.append(res)
    progress.update_progress()

print(f"cache {LocalCache.get_counter('cache_miss')} vals, used {LocalCache.get_counter('cache_find')} times")

for r in results:
    print(r['acc'])
    for bot in r['bots']:
        print(f"    profit={str(bot['profit_p']) + '%':<6} op={bot['op']:<4} \"{str(bot['config'])}\" \"{str(bot['last_conf'])}\"")
# print(results)


# # Вывод результатов или их дальнейшая обработка
# sorted_results = sorted(results, key=lambda x: (x['config'].ticker, x['config'].pretest_type, -float(x['profit_p'])), reverse=False)
# 
# last_val = ''
# for item in sorted_results:
#     cur_val = item['config'].ticker
#     if cur_val != last_val:
#         last_val = cur_val
#         print()
#     print(item)

Запуск в 14:40
Закончено в 04:18, количество 3, длительность 13:38:10                                              
cache 1605500 vals, used 4927600 times
{'profit': -0.47, 'p_avg': -0.0, 'config': t2 [0] |u0.2 d0.0| }
    profit=-13.27% op=2819 "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 5/fan6:0/2 x l84 x 1.0¤ "
    profit=-11.71% op=3926 "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 4/fan4:0/2 x l70 x 0.8(+x0.3)¤ |u0.01 d0.0| "
{'profit': -8.59, 'p_avg': -0.02, 'config': t3 [0] |u0.3 d0.0| }
    profit=-7.47% op=2829 "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 5/fan5:0/2 x l89 x 1.0¤ "
    profit=-13.13% op=3972 "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 6/fan4:0/2 x l46 x 0.8¤ "
{'profit': -12.06, 'p_avg': -0.03, 'config': t5 [0] |u0.5 d0.0| }
    profit=-14.53% op=3149 "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 6/fan3:0/2 x l69 x 1.0¤ "
    profit=-12.55% op=3891 "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 4/fan7:0/2 x l69 x 0.8(+x0.3)¤ |u0.01 d0.0| "


In [21]:
for r in results:
    print(r['acc'])
    for bot in r['bots']:
        print(f"    profit={str(bot['profit_p']) + '%':<6} op={bot['op']:<4} \"{str(bot['config'])}\" \"{str(bot['last_conf'])}\"")


{'profit': 0.75, 'p_avg': 0.03, 'config': t2 [0] |u0.2 d0.0| }
    profit=1.97%  op=397  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 6/fan4:0/2 x l82 x 1.0¤ "
    profit=3.15%  op=670  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 6/fan4:0/2 x l54 x 0.8¤ "
{'profit': 0.72, 'p_avg': 0.02, 'config': t2 [0] |u0.25 d0.0| }
    profit=2.22%  op=414  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 6/fan4:0/2 x l82 x 1.0¤ "
    profit=2.77%  op=683  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 6/fan4:0/2 x l54 x 0.8¤ "
{'profit': 0.85, 'p_avg': 0.03, 'config': t3 [0] |u0.3 d0.0| }
    profit=2.24%  op=414  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 6/fan4:0/2 x l82 x 1.0¤ "
    profit=2.67%  op=670  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "RNFT+ 6/fan4:0/2 x l54 x 0.8¤ "
{'profit': 1.99, 'p_avg': 0.07, 'config': t3 [0] |u0.35 d0.0| }
    profit=2.3%   op=413  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ " "MVID+ 6/fan4:0/2 x l82 x 1.0¤ "
    profit=3.25%  op=667  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2

За 1 день 24 окт

    {'profit': 0.25, 'config': auto_05 [0] |u0.5 d0.0| }
        profit=0.24% op=20   "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=0.3%  op=41   "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=0.19% op=38   "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "


[[{'exp': 'MVID fan F', 'profit': 240, 'profit_p': 0.24, 'profit_p_avg': 0.24, 'config': MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F , 'last_conf': MVID+ 10/fan7:0/2 x l48 x 0.6(+x0.2)¤ F , 'op': 20}, 

{'exp': 'RNFT fan F', 'profit': 303, 'profit_p': 0.3, 'profit_p_avg': 0.3, 'config': RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F , 'last_conf': RNFT+ 8/fan7:0/2 x l40 x 0.6(+x0.2)¤ F , 'op': 41}, 

{'exp': 'MTLR fan F', 'profit': 193, 'profit_p': 0.19, 'profit_p_avg': 0.19, 'config': MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F , 'last_conf': MTLR+ 10/fan7:0/2 x l48 x 0.4¤ |u0.01 d0.0| F , 'op': 38}]]

Целевые показатели
====

длительность 0:01:00                                               
cache 1502 vals, used 1568 times

[[{'exp': 'MVID fan ', 'profit': 1049, 'profit_p': 1.05, 'profit_p_avg': 0.04, 'config': MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ , 'last_conf': MVID+ 10/fan7:0/2 x l46 x 0.6(+x0.2)¤ , 'op': 285}, 

{'exp': 'RNFT fan ', 'profit': 1930, 'profit_p': 1.93, 'profit_p_avg': 0.06, 'config': RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ , 'last_conf': RNFT+ 8/fan7:0/2 x l40 x 0.6(+x0.2)¤ , 'op': 426}, 

{'exp': 'MTLR fan ', 'profit': 131, 'profit_p': 0.13, 'profit_p_avg': 0.0, 'config': MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| , 'last_conf': MTLR+ 10/fan7:0/2 x l46 x 0.4¤ |u0.01 d0.0| , 'op': 548}]]


    {'profit': 0.53, 'p_avg': 0.02, 'config': t1 [0] |u0.1 d0.0| }
        profit=1.42%  op=320  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=2.21%  op=436  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=1.66%  op=620  "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "
    {'profit': 1.52, 'p_avg': 0.05, 'config': t2 [0] |u0.2 d0.0| }
        profit=1.33%  op=310  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=2.07%  op=454  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=1.12%  op=646  "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "
    {'profit': 1.77, 'p_avg': 0.06, 'config': t3 [0] |u0.3 d0.0| }
        profit=1.43%  op=327  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=2.24%  op=457  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=1.31%  op=665  "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "
    {'profit': 1.78, 'p_avg': 0.06, 'config': t5 [0] |u0.5 d0.0| }
        profit=1.48%  op=344  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=2.03%  op=463  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=1.36%  op=675  "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "
    {'profit': 1.78, 'p_avg': 0.06, 'config': t5 [0] |u0.5 d0.0| }
        profit=1.48%  op=344  "MVID+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=2.03%  op=463  "RNFT+ 4/fan7:0/2 x l1 x 0.6(+x0.2)¤ F "
        profit=1.36%  op=675  "MTLR+ 4/fan7:0/2 x l1 x 0.6¤ |u0.01 d0.0| F "